hugolib/page.go

```diff
@@ -937,5 +953,12 @@ func (p *Page) TargetPath() (outfile string) {
 		outfile = helpers.ReplaceExtension(p.Source.LogicalName(), p.Extension())
 	}
 
-	return filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile))
+	return p.addLangPrefix(filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile)))
+}
+
+func (p *Page) addLangPrefix(outfile string) string {
+	if p.lang == "" {
+		return outfile
+	}
+	return p.lang + "/" + outfile
```

Comment:
I like the general approach.
On Tue, Dec 29, 2015 at 11:11 PM Cameron Moore <notifications@github.com>
wrote:

> In hugolib/page.go
> <https://github.com/spf13/hugo/pull/1734#discussion_r48586001>:
>
> > @@ -937,5 +953,12 @@ func (p *Page) TargetPath() (outfile string) {
> >  		outfile = helpers.ReplaceExtension(p.Source.LogicalName(), p.Extension())
> >  	}
> >
> > -	return filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile))
> > +	return p.addLangPrefix(filepath.Join(strings.ToLower(helpers.MakePath(p.Source.Dir())), strings.TrimSpace(outfile)))
> > +}
> > +
> > +func (p *Page) addLangPrefix(outfile string) string {
> > +	if p.lang == "" {
> > +		return outfile
> > +	}
> > +	return p.lang + "/" + outfile
>
> On line 940, we take a url-style path and converts it to an OS dependent
> path with filepath.FromSlash. filepath.FromSlash will convert
> posts/index.html to posts\index.html on Windows.
>
> @bep <https://github.com/bep> is correct. In this context, you want to
> use filepath.Join.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spf13/hugo/pull/1734/files#r48586001>.
>